src/scheduler/workers/types.ts

```diff
@@ -0,0 +1,19 @@
+import { List } from "@prisma/client";
+
+export interface Meta {
+  /**
+   * annual review reference for list (`List.jsonData.currentAnnualReview.reference`)
+   */
+  reference: string;
+  weeksUntilUnpublish?: number;
+  daysUntilUnpublish?: number;
+  weeksSinceStart: number;
+  parsedUnpublishDate: string;
+  countryName: string;
+}
```

Comment:
for the weekly reminders, `weeksUntilUnpublish` is required.

Same goes for day before reminder, it should also be required for their respective tasks. 

Revert or refactor please. 

but I think it's fine to have this duplicated in their respective "namespaces". (weekly vs day before etc). 

If you want a "unified" Meta type, use union types. But there's no instance where meta would be shared between the different tasks. 


```suggestion


export interface WeeklyMeta {
  /**
   * annual review reference for list (`List.jsonData.currentAnnualReview.reference`)
   */
  reference: string;
  weeksUntilUnpublish: number;
  weeksSinceStart: number;
  parsedUnpublishDate: string;
  countryName: string;
}


export interface DayBeforeMeta { 

  /**
   * annual review reference for list (`List.jsonData.currentAnnualReview.reference`)
   */
  reference: string;
  daysUntilUnpublish: number;
  weeksSinceStart: number;
  parsedUnpublishDate: string;
  countryName: string;

}


export interface Meta = WeeklyMeta | DayBeforeMeta;
```

with union types, when consuming `Meta` for example, the type will be "WeeklyMeta" and "DayBeforeMeta" - and ts will be able to differentiate the two because they have overlapping properties and properties that don't.

Reply:
refactored to move to their respective namespaces rather than using union type.